### PR TITLE
Meyer3cj codereview

### PR DESCRIPTION
**Pull request by meyer3cj**
•	Analysis of the program.
I think this is a good idea when looking at the design document. There is not a lot of functionality as of yet. The program is coded cleanly.
•	Was the program available in UC Github on time?
The program was available in time.
•	Is the program easy for you, as an outsider, to understand?
the program is not very easy to understand in my opinion, The only thing that I wish had more explanation was the script. I know the script generates the calendar.
•	Does the program compile?
Yes. The program compiles and tests pass.
Three specific technical concepts that I learned

1. I added an endpoint to the controller to contain the metrics page.
2. I added Javadoc to the methods in the service classes.
3. I added a form to the metrics page to take a user entry.

### CODE REVIEW

File: src/main/java/com/moodybluez/enterprise/MoodyBluezController.java [hunk 1]
```diff
@@ -14,4 +14,9 @@
     public String index() {
         return "index";
     }
+    @RequestMapping("/metrics")
+    public String metrics() {
+        return "metrics";
+    }
+
```
Comment (meyer3cj) [inline src/main/java/com/moodybluez/enterprise/MoodyBluezController.java:21]: I added an endpoint to house the metrics endpoint.

File: src/main/java/com/moodybluez/enterprise/dto/Date.java [hunk 2]
```diff
@@ -2,9 +2,10 @@
...
 import lombok.Data;
 
+
 public @Data
 class Date {
     private String date;
-    private int dayOfWeekID;
-    private String dayOfWeekDesc;
+    private int dayID;
+    private String dayDesc;
```
Comment (meyer3cj) [inline src/main/java/com/moodybluez/enterprise/dto/Date.java:10]: I made the variable names more general for understandability.

File: src/main/java/com/moodybluez/enterprise/service/EntryServiceStub.java [hunk 3]
```diff
@@ -20,6 +20,12 @@ public void saveEntry(Entry entry) {
         moodEntryMap.put(entry.date.getDate(), entry);
     }
 
+    /**
+     *
+     * @param date
+     * @return
+     * The method fetches the entries by date
+     */
```
Comment (meyer3cj) [inline src/main/java/com/moodybluez/enterprise/service/EntryServiceStub.java:28]: I added Javadoc for classes to let other reviewers know what the method does.

File: src/main/java/com/moodybluez/enterprise/service/EntryServiceStub.java [hunk 4]
```diff
@@ -39,6 +45,11 @@ public Entry fetchByDate(String date) {
         }
     }
 
+    /**
+     *
+     * @return
+     * The method returns all entries from the hash map
+     */
```
Comment (meyer3cj) [inline src/main/java/com/moodybluez/enterprise/service/EntryServiceStub.java:52]: I added Javadoc for classes to let other reviewers know what the method does.

File: src/main/java/com/moodybluez/enterprise/service/MoodServiceStub.java [hunk 5]
```diff
@@ -5,6 +5,12 @@
 
 @Component
 public class MoodServiceStub implements IMoodService {
+    /**
+     *
+     * @param id unique identifier for a mood
+     * @return
+     * fetches the mood by id
+     */
```
Comment (meyer3cj) [inline src/main/java/com/moodybluez/enterprise/service/MoodServiceStub.java:13]: I added Javadoc for classes to let other reviewers know what the method does.

File: src/main/resources/templates/fragments/topnav.html [hunk 6]
```diff
@@ -5,10 +5,10 @@
             <div class="collapse navbar-collapse" id="navbarNav">
                 <ul class="navbar-nav">
                     <li class="nav-item">
-                        <a class="nav-link active" aria-current="page" href="#">Home</a>
+                        <a class="nav-link active" aria-current="page" href="/">Home</a>
```
Comment (meyer3cj) [inline src/main/resources/templates/fragments/topnav.html:8]: I changed the path to make the navbar functional.

File: src/main/resources/templates/fragments/topnav.html [hunk 7]
```diff
@@ -5,10 +5,10 @@
...
                     </li>
                     <li class="nav-item">
-                        <a class="nav-link" href="#">Metrics</a>
+                        <a class="nav-link" href="/metrics">Metrics</a>
```
Comment (meyer3cj) [inline src/main/resources/templates/fragments/topnav.html:11]: I set the metrics endpoint to work on the navbar.

File: src/main/resources/templates/index.html [hunk 8]
```diff
@@ -16,9 +16,7 @@
     </head>
 <body>
 
-<div th:replace="fragments/topnav :: navbar">
-
-</div>
+<div th:replace="fragments/topnav :: navbar"></div>
```
Comment (meyer3cj) [inline src/main/resources/templates/index.html:19]: There was a small formatting issue I changed to make the code easier to read.

File: src/test/java/com/moodybluez/enterprise/EnterpriseApplicationTests.java [hunk 9]
```diff
@@ -81,9 +81,9 @@ private void whenEntryIsCompleted() {
 		entry.setReasonDesc(reasonForMood);
 		entry.setDate(new Date());
 		entry.date.setDate(entryDate);
-		entry.date.setDayOfWeekID(dayOfWeek.getValue());
-		entry.date.setDayOfWeekDesc(dayOfWeek.name());
-		entry.date.setDayOfWeekID(dayOfWeek.getValue());
+		entry.date.setDayID(dayOfWeek.getValue());
+		entry.date.setDayDesc(dayOfWeek.name());
+		entry.date.setDayID(dayOfWeek.getValue());
```
Comment (meyer3cj) [inline src/test/java/com/moodybluez/enterprise/EnterpriseApplicationTests.java:86]: I changed variables in the classes, so I had to change them in the unit test also.

File: src/main/resources/templates/metrics.html [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+<body>
+
+<div th:replace="fragments/topnav :: navbar"></div>
+
+<form action="/">
+    <input class="form-control form-control-lg"type="text" id="date" name="date" placeholder="Date:"><br>
+    <input class="form-control form-control-lg"type="text" id="mood" name="mood" placeholder="Mood:"><br>
+    <input class="form-control form-control-lg"type="text" id="description" name="description" placeholder="Description"><br>
+    <input type="submit" value="Submit">
+</form>
```
Comment (meyer3cj) [inline src/main/resources/templates/metrics.html:26]: This is the metrics page I added. The page contains a form which goes along with the design document. I suggest to the group that when a user clicks the calendar, they send the user to the metrics page and have the date form have the date filled out and have the user complete the form before they record their mood.

File: src/main/java/com/moodybluez/enterprise/MoodyBluezController.java (hunk 1 above)
Comment (discospiff) [reply src/main/java/com/moodybluez/enterprise/MoodyBluezController.java:21]: Is there a matching HTML page?   If not, this is incomplete functionality.

File: src/main/java/com/moodybluez/enterprise/dto/Date.java (hunk 2 above)
Comment (discospiff) [reply src/main/java/com/moodybluez/enterprise/dto/Date.java:10]: There's a difference between day and day of week... I'd keep this as is.  Day of month, day of year, etc...

File: src/main/java/com/moodybluez/enterprise/service/EntryServiceStub.java (hunk 3 above)
Comment (discospiff) [reply src/main/java/com/moodybluez/enterprise/service/EntryServiceStub.java:28]: Complete the JavaDoc to make it useful.   What are the params?  Returns?

File: src/main/resources/templates/metrics.html (hunk 10 above)
Comment (discospiff) [reply src/main/resources/templates/metrics.html:26]: Ok, here is the page.

**Comment by discospiff** [comment]
There's a bit of new functionality... not a as much tech debt reduction.  I recommend the team look at each change and decide what to keep, what to pitch, and what to expand on and merge.

**Pull request closed without merging**